static/js/realm_playground.js

```diff
@@ -28,18 +31,41 @@ export function sort_pygments_pretty_names_by_priority(generated_pygments_data)
     const priority_sorted_pygments_data = Object.keys(generated_pygments_data.langs).sort(
         typeahead_helper.compare_by_popularity,
     );
-    for (const data of priority_sorted_pygments_data) {
-        const pretty_name = generated_pygments_data.langs[data].pretty_name;
-        // JS maintains the order of insertion in a set so we don't need to worry about
-        // the priority changing.
-        if (!pygments_pretty_name_list.has(pretty_name)) {
-            pygments_pretty_name_list.add(pretty_name);
+    for (const alias of priority_sorted_pygments_data) {
+        const pretty_name = generated_pygments_data.langs[alias].pretty_name;
+        // JS Map remembers the original order of insertion of keys.
+        if (map_pygments_pretty_name_to_aliases.has(pretty_name)) {
+            map_pygments_pretty_name_to_aliases.get(pretty_name).push(alias);
+        } else {
+            map_pygments_pretty_name_to_aliases.set(pretty_name, [alias]);
         }
     }
 }
 
-export function get_pygments_pretty_names_list() {
-    return Array.from(pygments_pretty_name_list);
+export function get_pygments_typeahead_list(query) {
+    const lookup_table = new Map();
+    const pygments_pretty_name_list = [];
+
+    // Adds a typeahead that allows selecting a custom language, by adding a
+    // "Custom language" label in the first position of the typeahead list.
+    const clean_query = typeahead.clean_query_lowercase(query);
+    if (clean_query !== "") {
+        pygments_pretty_name_list.push(clean_query);
+        lookup_table[clean_query] = $t(
```

Comment:
This is wrong; a `Map` needs to be indexed with `.get()` and `.set()`, not with `[]`.